kernel/src/hil/time.rs

```diff
@@ -8,6 +10,15 @@ pub trait Time {
 
     /// Returns whether a timer or alarm is currently armed
     fn is_armed(&self) -> bool;
+
+    /// Returns the current time in hardware clock units.
+    fn now(&self) -> u32;
+}
+
+pub trait Counter: Time {
+    fn start(&self) -> ReturnCode;
+    fn stop(&self) -> ReturnCode;
```

Comment:
If I grok supertraits correctly, the result of this will be an object that has both `stop` and `disable` methods (assuming both traits have been `use`d). I don't necessarily love redundant names like `stop_counter`, but I'm concerned this interface could be ambiguous.

Reply:
Yes. You start and stop a counter, but enable/disable a timer. It's not ambiguous, although it might be unclear. We could go with stop_counter and disable_timer if you want.

Reply:
What do `my_counter.disable()` and `my_counter.is_armed()` do? Are they synonyms for `stop()` and `is_running()`?

Reply:
I agree with Brad's concern. By breaking Counter up into Time and Counter, there's this weird inconsistency.  They should not be synonyms for `stop()` and `is_running()`, because they are about a callback.